core/primitives/src/epoch_manager.rs

```diff
@@ -77,35 +77,54 @@ impl ShardConfig {
 
 #[derive(Clone)]
 pub struct AllEpochConfig {
+    use_production_config: bool,
     genesis_epoch_config: EpochConfig,
-    simple_nightshade_epoch_config: EpochConfig,
+    simple_nightshade_shard_config: Option<ShardConfig>,
 }
 
 impl AllEpochConfig {
     pub fn new(
+        use_production_config: bool,
         genesis_epoch_config: EpochConfig,
         simple_nightshade_shard_config: Option<ShardConfig>,
     ) -> Self {
-        let mut config = genesis_epoch_config.clone();
-        if let Some(ShardConfig {
-            num_block_producer_seats_per_shard,
-            avg_hidden_validator_seats_per_shard,
-            shard_layout,
-        }) = simple_nightshade_shard_config
-        {
-            config.num_block_producer_seats_per_shard = num_block_producer_seats_per_shard;
-            config.avg_hidden_validator_seats_per_shard = avg_hidden_validator_seats_per_shard;
-            config.shard_layout = shard_layout;
-        }
-        Self { genesis_epoch_config, simple_nightshade_epoch_config: config }
+        Self { use_production_config, genesis_epoch_config, simple_nightshade_shard_config }
     }
 
-    pub fn for_protocol_version(&self, protocol_version: ProtocolVersion) -> &EpochConfig {
+    pub fn for_protocol_version(&self, protocol_version: ProtocolVersion) -> EpochConfig {
+        let mut config = self.genesis_epoch_config.clone();
         if checked_feature!("stable", SimpleNightshade, protocol_version) {
-            &self.simple_nightshade_epoch_config
-        } else {
-            &self.genesis_epoch_config
-        }
+            if let Some(ShardConfig {
```

Comment:
comment (that in case of SimpleNighshade protocol version, we overwrite the fields).